workers/generic-worker/main.go

```diff
@@ -584,52 +583,25 @@ func ClaimWork() *TaskRun {
 
 func (task *TaskRun) validatePayload() *CommandExecutionError {
 	jsonPayload := task.Definition.Payload
-	defaults.SetDefaults(&task.Payload)
-	err := json.Unmarshal(jsonPayload, &task.Payload)
-	if err != nil {
-		return MalformedPayloadError(err)
+	validateErr := task.validateJSON(jsonPayload, JSONSchema())
+	if validateErr != nil {
+		return validateErr
 	}
-	schemaLoader := gojsonschema.NewStringLoader(JSONSchema())
-	docLoader := gojsonschema.NewStringLoader(string(jsonPayload))
-	result, err := gojsonschema.Validate(schemaLoader, docLoader)
+	payload := map[string]interface{}{}
+	err := json.Unmarshal(jsonPayload, &payload)
 	if err != nil {
 		return MalformedPayloadError(err)
```

Comment:
I think this would need to be an internal error - maybe you even need to panic if there is an error here. At this point, you have ensured that it complies to the payload schema, but then it cannot be unmarshaled into a map[string]interface{} - which would be a very serious error, and not because of a malformed payload.